pkg/machinery/config/generate/worker.go

```diff
@@ -74,10 +74,17 @@ func (in *Input) worker() ([]config.Document, error) {
 		machine.MachineFeatures.DiskQuotaSupport = pointer.To(true)
 	}
 
-	if in.Options.KubePrismPort > 0 {
+	if in.Options.KubePrismPort.IsPresent() {
+		if kubePrismPort := in.Options.KubePrismPort.ValueOrZero(); kubePrismPort > 0 {
```

Comment:
This broke my head. I just realized that we use `KubePrismPort.Value == 0` as a signal value, but `ValueOrZero` will also return `0` if there is nothing inside, and the thing protecting us from that is `in.Options.KubePrismPort.IsPresent()` expression above.

I actually think that you can do better here:
```suggestion
	if kubePrismPort, present := in.Options.KubePrismPort.Get(); present {
		if kubePrismPort > 0 {
```

This is both shorter and clearer IMHO.

Reply:
I thought of that, but it looks less explicit to me, but :shrug:  not a big deal